src/adm/settings.rs

```diff
@@ -176,12 +177,56 @@ where
 }
 
 #[derive(Debug, Default, Clone)]
-pub struct AdmSettings {
+pub struct AdmPse {
+    pub partner_id: String,
+    pub sub1: String,
+    pub endpoint: String,
+}
+
+/// ADM Partner/Sub1/Endpoint.
+/// These change depending on the type of device requesting the tile.
+impl AdmPse {
+    /// Return the information for a mobile connection
+    pub fn as_mobile(settings: &Settings) -> Self {
+        let default = Self::as_default(settings);
+        AdmPse {
+            partner_id: settings
+                .adm_mobile_partner_id
+                .clone()
+                .unwrap_or(default.partner_id),
+            sub1: settings.adm_mobile_sub1.clone().unwrap_or(default.sub1),
+            endpoint: settings
+                .adm_mobile_endpoint_url
+                .clone()
+                .unwrap_or(default.endpoint),
+        }
+    }
+
+    /// Return the information for a non-mobile connection
+    pub fn as_default(settings: &Settings) -> Self {
```

Comment:
Thanks! I was debating the best way of dealing with this.
I also didn't really want to keep pumping stuff into ServerState if we didn't have to. This is probably a bit of my "if 2 than _n_" thinking, but i figure setting this pattern makes adding future ones a bit easier to deal with. Likewise, the clone I'm doing is just for the specific values, not the entire settings, so that should be reasonably lighter weight.